src/retry/retryUtility.ts

```diff
@@ -57,40 +51,42 @@ export async function execute({
   }
   if (!retryPolicies) {
     retryPolicies = {
-      endpointDiscoveryRetryPolicy: new EndpointDiscoveryRetryPolicy(globalEndpointManager, request.operationType),
+      endpointDiscoveryRetryPolicy: new EndpointDiscoveryRetryPolicy(
+        globalEndpointManager,
+        requestContext.operationType
+      ),
       resourceThrottleRetryPolicy: new ResourceThrottleRetryPolicy(
         connectionPolicy.retryOptions.maxRetryAttemptCount,
         connectionPolicy.retryOptions.fixedRetryIntervalInMilliseconds,
         connectionPolicy.retryOptions.maxWaitTimeInSeconds
       ),
       sessionReadRetryPolicy: new SessionRetryPolicy(
         globalEndpointManager,
-        request.resourceType,
-        request.operationType,
+        requestContext.resourceType,
+        requestContext.operationType,
         connectionPolicy
       ),
-      defaultRetryPolicy: new DefaultRetryPolicy(request.operationType)
+      defaultRetryPolicy: new DefaultRetryPolicy(requestContext.operationType)
     };
   }
-  const httpsRequest = createRequestObjectFunc(connectionPolicy, requestOptions, body);
```

Comment:
@christopheranderson this change one line but significant. Before we were creating this promise up here and awaiting inside the try block on line 89. With abort controller, this caused unhandled promise errors. So I moved it down to line 89, but does that change all the location routing stuff? I could move that into the try block to

Reply:
Resolved this on v3 already in the RequestContext PR